client/shared/src/main/scala/org/http4s/client/DefaultClient.scala

```diff
@@ -88,14 +87,13 @@ private[http4s] abstract class DefaultClient[F[_]](implicit F: MonadCancelThrow[
   def streaming[A](req: F[Request[F]])(f: Response[F] => Stream[F, A]): Stream[F, A] =
     Stream.eval(req).flatMap(stream).flatMap(f)
 
+  private def reqWithMediaRangeAndQValue[A](req: Request[F], d: EntityDecoder[F, A]) =
+    d.consumes.toList.toNel.fold(req)(m => req.addHeader(Accept(m.map(MediaRangeAndQValue(_)))))
```

Comment:
This is probably a few femtoseconds slower than the original, but since we're about to make an HTTP request, it is probably completely imperceptible.  And I like getting rid of the unsafe methods.  And if someone wants to reoptimize this, now it's in one place.

Reply:
> `.toList.toNel`

In the capacity of a grumpy micro-benchmarker of certain parts of http4s, this will also consume a bit more heap and stress out the GC (in terms of femtoseconds). Unsafe API here is a reasonable choice since we have checks on non-emptiness. 

But, I really like this refactoring into the dedicated `def` 👍🏻

Reply:
> Which part is it that's expected to add a bit more heap / GC / execution time? Would this be an improvement?
> 
> ```scala
>   private def reqWithMediaRangeAndQValue[A](req: Request[F], d: EntityDecoder[F, A]) =
>     d.consumes.toList match {
>       case head :: next =>
>         req.addHeader(Accept(NonEmptyList(head, next).map(MediaRangeAndQValue(_))))
>       case Nil => req
>     }
> ```

I think this would be an improvement. It avoids allocating the extra `Option` that using `toNel` had.
The original implementations avoiding calling `toList` if the `Set` was empty, but doing that is just going to return `Nil` if the set is empty anyways. It's fine.

@Adam-McDevitt, if you're still willing, yes I think your suggestion here would be an improvement.